packages/backend/src/open_payments/payment/incoming/routes.test.ts

```diff
@@ -421,4 +424,171 @@ describe('Incoming Payment Routes', (): void => {
+            },
+            description: incomingPayment3.description,
+            expiresAt: expiresAt.toISOString(),
+            receivedAmount: {
+              value: '0',
+              assetCode: asset.code,
+              assetScale: asset.scale
+            },
+            externalRef: '#213',
+            state: IncomingPaymentState.Pending.toLowerCase()
+          }
+        ]
+      }
+    )
+    test.each`
+      id              | headers                     | first           | last            | cursor          | status | message                           | description
+      ${'not_a_uuid'} | ${null}                     | ${'10'}         | ${null}         | ${null}         | ${400} | ${'invalid account id'}           | ${'invalid account id'}
+      ${null}         | ${{ Accept: 'text/plain' }} | ${'10'}         | ${null}         | ${null}         | ${406} | ${'must accept json'}             | ${'invalid Accept header'}
+      ${null}         | ${null}                     | ${['10', '20']} | ${null}         | ${null}         | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${null}         | ${['10', '20']} | ${null}         | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${'hello'}      | ${null}         | ${null}         | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${null}         | ${null}         | ${['a', 'b']}   | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${'10'}         | ${'10'}         | ${null}         | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${null}         | ${'10'}         | ${undefined}    | ${400} | ${'invalid pagination paramters'} | ${'invalid pagination paramters'}
+      ${null}         | ${null}                     | ${null}         | ${null}         | ${'not_a_uuid'} | ${400} | ${'invalid cursor'}               | ${'invalid cursor'}
+    `(
+      'returns $status on $description',
+      async ({
+        id,
+        headers,
+        first,
+        last,
+        cursor,
+        status,
+        message
+      }): Promise<void> => {
+        const params = id
+          ? { accountId: id }
+          : { accountId: incomingPayment.accountId }
+        const query = {
+          cursor: cursor === null ? incomingPayment.id : cursor
+        }
+        if (first) query['first'] = first
+        if (last) query['last'] = last
+        const ctx = setup({ headers, query }, params)
+        await expect(incomingPaymentRoutes.list(ctx)).rejects.toMatchObject({
+          status,
+          message
+        })
+      }
+    )
+
+    test.each`
```

Comment:
:thinking: maybe we should do this for routes tests:
- https://github.com/interledger/rafiki/pull/242